File: model-microservice/map/helper/map.py
import os
import requests
from dotenv import load_dotenv,find_dotenv
from libs.logger import logger

load_dotenv(find_dotenv())

GOOGLE_MAPS_API_KEY=os.getenv('GOOGLE_MAPS_API_KEY')

def get_hostel():
    lat = 27.65639942349444
    lng = 85.35119061937837
    radius = 9000
    
    try:
        response = requests.get(
            'https://maps.googleapis.com/maps/api/place/nearbysearch/json',
            params={
                'location': f'{lat},{lng}',
                'radius': radius,
                'type': 'lodging',  
                'keyword': 'hostel', 
                'key': GOOGLE_MAPS_API_KEY
            }
        )
        data = response.json().get('results', [])
        return data


    except Exception as e:
        logger.error(f'Error in getting hostel{e}')
        return None
        

File: node-microservice/src/helper/maps/map.ts
import axios from 'axios';

const GOOGLE_MAPS_API_KEY = 'AIzaSyChRHG8gb0TwMq2YOdf_djXNkDxtokdAJI';

export const getHotels = async (lat: any, lng: any, radius: any) => {
  try {
    const response = await axios.get(
      'https://maps.googleapis.com/maps/api/place/nearbysearch/json',
      {
        params: {
          location: `${lat},${lng}`,
          radius: radius,
          type: 'lodging',
          keyword: 'hostel',
          key: GOOGLE_MAPS_API_KEY,
        },
      }
    );
    return response.data.result;
  } catch (err) {
    throw new Error('Google Map Error');
  }
};
